R/pivot_wider_pknca_results.R

```diff
@@ -137,15 +145,22 @@ pivot_wider_pknca_results <- function(myres) {
   if (length(unique_values) == 0) NA_character_ else paste(unique_values, collapse = ", ")
 }
 
-#' Helper function to add "label" attribute to columns based on parameter names
+#' Helper function to add "label" attribute to columns based on parameter names.
 #' @noRd
-.add_label_attribute <- function(df, myres) {
-
+#' @keywords internal
+add_label_attribute <- function(df, myres) {
   mapping_vr <- myres$result %>%
-    mutate(PPTESTCD_unit = ifelse(PPSTRESU != "", paste0(PPTESTCD, "[", PPSTRESU, "]"), PPTESTCD),
-           PPTESTCD_cdisc = gsub("\\$", "", translate_terms(PPTESTCD,
-                                                            mapping_col = "PPTESTCD",
-                                                            target_col = "PPTEST"))) %>%
+    mutate(
+      PPTESTCD_unit = case_when(
+        type_interval == "manual" ~ paste0(
+          PPTESTCD, "_", start, "-", end,
+          ifelse(PPSTRESU != "", paste0("[", PPSTRESU, "]"), "")
+        ),
+        PPSTRESU != "" ~ paste0(PPTESTCD, "[", PPSTRESU, "]"),
+        TRUE ~ PPTESTCD
+      ),
+      PPTESTCD_cdisc = translate_terms(PPTESTCD, mapping_col = "PPTESTCD", target_col = "PPTEST")
+    ) %>%
     select(PPTESTCD_cdisc, PPTESTCD_unit) %>%
     distinct() %>%
     pull(PPTESTCD_cdisc, PPTESTCD_unit)
```

Comment:
suggestion: Just a refactoring to
1) simplify the nested/implicit `ifelse`s in the `case_when`
2) to rename variable `PPTESTCD_cdisc` > `PPTEST_cdisc`

```suggestion
    mutate(
      PPSTRESU = ifelse(
        PPSTRESU != "",
        paste0("[", PPSTRESU, "]"),
        ""
      ),
      PPTESTCD = ifelse(
        type_interval == "manual",
        paste0(PPTESTCD, "_", start, "-", end),
        PPTESTCD
      ),
      PPTESTCD_unit = paste0(PPTESTCD, PPSTRESU),
      PPTEST_cdisc = translate_terms(PPTESTCD, mapping_col = "PPTESTCD", target_col = "PPTEST")
    ) %>%
    select(PPTEST_cdisc, PPTESTCD_unit) %>%
    distinct() %>%
    pull(PPTEST_cdisc, PPTESTCD_unit)
```